datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -89,6 +89,9 @@ def __init__(self, *args, **kwargs):
                 # new-style init: the 3rd argument is `instances`
                 self.instances = args[2]
 
+        # Agent 6+ will only have one instance
```

Comment:
Because there is a new AgentCheck instantiated for every check instance cc @olivielpeau or @truthbk maybe can confirm

Reply:
Confirmed

Reply:
this is correct, since v6.0 the Agent instantiates one instance of the `AgentCheck` class per configured instance of the check. Relevant code: https://github.com/DataDog/datadog-agent/blob/6.9.0/pkg/collector/py/loader.go#L243-L255

Reply:
ok i wasn't aware of this. Thanks for the update.